src/common.rs

```diff
@@ -37,6 +37,18 @@ impl ColorType {
             _ => None,
         }
     }
+
+    pub(crate) fn raw_row_length_from_width(&self, depth: BitDepth, width: u32) -> usize {
+        let bits = width as usize * self.samples() * depth as usize;
+
+        let extra = bits % 8;
+        bits / 8
+            + match extra {
+                0 => 0,
+                _ => 1,
+            }
```

Comment:
This is a convoluted way to say `(bits + 7) / 8`

Reply:
It's the overflow safe way. But actually all of the computation of buffer size disregards pointer size and overflow in any case already.. Eh.